# https://github.com/lmcinnes/umap/commit/1412b49fddf5c9b948fa57f40ba3973137578ac2
# It seems like previous Numba versions (up to 0.51.2 or 0.52.0) may have
# worked 'by accident'. When running umap against a recent
# development version of Numba.
diff --git a/umap/layouts.py b/umap/layouts.py
index ebb9e8e2..c3d79fc7 100644
--- a/umap/layouts.py
+++ b/umap/layouts.py
@@ -33,7 +33,7 @@ def clip(val):
     locals={
         "result": numba.types.float32,
         "diff": numba.types.float32,
-        "dim": numba.types.int32,
+        "dim": numba.types.intp,
     },
 )
 def rdist(x, y):
